support/testing/run_device_batches.sh

```
@@ -0,0 +1,84 @@
+    local config_file="$2"
+    
+    echo -e "${YELLOW}📱 Running tests for batch $batch_num...${NC}"
+    
+    # Backup original config
+    cp "$ORIGINAL_CONFIG" "${ORIGINAL_CONFIG}.backup"
+    
+    # Use batch config
+    cp "$config_file" "$ORIGINAL_CONFIG"
+    
+    # Run tests
+    local start_time=$(date +%s)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)